merging client stuff to master

harbar20 opened the pull request.

harbar20 committed:
i think i'm almost done with qrgen.py, lmk on slack if there's anything else to add to it

harbar20 committed:
finished admin-side logging in stuff, now moving on to qr gen stuff

harbar20 committed:
now generates qr code, might add support for different sizes and printing directly as well.

harbar20 committed:
finished qrcode gen at last, now to make print button again

harbar20 committed:
deleted application.py, added .gitignore

harbar20 committed:
finished everything client-related except the bot

harbar20 committed:
bot pays snekcoin now, yay

lukec11 commented on main.py:
```diff
@@ -1 +1,183 @@
+@app.route("/qrcode", methods=["GET", "POST"])
+def qrcode():
+    #Generates the QR Code
+    gen = GPQRGen("http://localhost:3000", flask.session["amount"])
```
can you make this `0.0.0.0` instead of localhost? that will bind to all interfaces

harbar20 replied:
0.0.0.0 never works for me when i try to test using it and i have no idea why. Any ideas?

lukec11 replied:
I'm not sure whether `0.0.0.0` works on windows like it does on linux -
might be best to change it during testing? We need this for production
though so it can run properly within the docker container.

lukec11 commented on main.py:
```diff
@@ -1 +1,183 @@
+    return ("Done! Check Slack for your new gp!", 200)
+
+if __name__ == '__main__': #this checks that it's actually you running main.py not an import
+    app.run(host='0.0.0.0', debug=True, port=3000) #this tells flask to go
```
`debug=False` for production please lol

harbar20 committed:
changed debug to False, also changed QRCodeGen object's domain to 0.0.0.0

lukec11 committed:
switch to env vars and format

switches to environment variables instead of config.json + Format with autopep8

lukec11 committed:
update to env vars in qrgen

lukec11 committed:
remove broken reqs and add poetry

harbar20 committed:
added comments clarifying where users and admin start

lukec11 commented:
I'm not sure whether `0.0.0.0` works on windows like it does on linux -
might be best to change it during testing? We need this for production
though so it can run properly within the docker container.

On Wed, Jun 17, 2020 at 3:39 PM Harshith Iyer <notifications@github.com>
wrote:

> *@harbar20* commented on this pull request.
> ------------------------------
>
> In main.py
> <https://github.com/hackclub/hack-coins/pull/1#discussion_r441788470>:
>
> > +        logged_in = False
> +
> +    if request.method == 'POST':
> +        amount = request.form.get("amount")
> +        flask.session["amount"] = int(amount)
> +        return redirect("/qrcode")
> +
> +    if not logged_in:
> +        return render_template("adminlogin.html", clientId=client_id)
> +    else:
> +        return render_template("generate.html", name=username)
> +
> +@app.route("/qrcode", methods=["GET", "POST"])
> +def qrcode():
> +    #Generates the QR Code
> +    gen = GPQRGen("http://localhost:3000", flask.session["amount"])
>
> 0.0.0.0 never works for me when i try to test using it and i have no idea
> why. Any ideas?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/hack-coins/pull/1#discussion_r441788470>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADUENSVXIDFKBSOPVGJ2BKDRXELVNANCNFSM4OAFB7AQ>
> .
>

harbar20 committed:
idk what i'm doing, git is being weird

harbar20 committed:
uses softbank api to give all currencies

lukec11 commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+config.json
```
what is this file used for and why is it gitignored?

harbar20 replied:
It used to be where the airtable API keys and stuff used to go but you changed them to be env variables so I'll remove it.

lukec11 commented on main.py:
```diff
@@ -1 +1,194 @@
+app.secret_key = secretKey
+
+# Airtable bases
+claims_base = Airtable(base_key, "Coin Claims", api_key=airtable_auth_key)
+admin_base = Airtable(base_key, "Verified Generators",
```
Can we make both of these formatted the same?

harbar20 replied:
oh yeah I'll do that, idk why it's like that lmao

lukec11 commented on main.py:
```diff
@@ -1 +1,194 @@
+    })
+
+    # The bot gives all the currencies to the user
+    requests.post("https://softbank.codered.cloud/hackcoin/redeem", {
```
i don't know about this...hack club has officially endorsed GP, and I'm not sure it's worth introducing a dependency on an app like softbank.

Additionally, this isn't checked - if softbank (or codered) go down, hack-coins crashes.

harbar20 replied:
That's fair. But until banker and banker API is done, I'm using this unless you have any alternative.

lukec11 commented on pyproject.toml:
```diff
@@ -0,0 +1,20 @@
+[tool.poetry]
+name = "hack-coins"
+version = "0.1.0"
+description = ""
```
can we add a basic description?

harbar20 replied:
Sure, will add.

lukec11 commented on pyproject.toml:
```diff
@@ -0,0 +1,20 @@
+[tool.poetry]
+name = "hack-coins"
+version = "0.1.0"
```
should this be published as 0.1.0 or 1.0.0?

harbar20 replied:
I'm honestly not sure. What do you think I should publish it as?

lukec11 commented on qrgen.py:
```diff
@@ -0,0 +1,37 @@
+        record = {
+            "UUID": self.newUUID,
+            "Amount": amount,
+            "Claimant Slack Email": ""
```
I don't have a lot of experience with the python airtable library - do you need to insert this key, or can you leave it out? If you can leave it out it seems like unnecessary cruft here

lukec11 replied:
might be a better idea to pass in the claimant's slack email or whatever to the class here

lukec11 commented on templates/generate.html:
```diff
@@ -0,0 +1,12 @@
+        <title>Hack Coins | Admin Generate QR Code</title>
+    </head>
+    <body>
+        <h2>Welcome, @{{ name }}!</h2>
```
is {{ name }} necessarily going to be fulfilled by their slack ID? 

If it is just resolve this

lukec11 commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+config.json
+__pycache__
```
add `static/`

lukec11 reviewed (commented):
This looks quite good! Tiny tweaks before merge.

There are also a couple of files in the `static` folder, that should be gitignored and the files removed

harbar20 committed:
removed static folder

harbar20 committed:
added static folder to .gitignore

harbar20 committed:
added / to the directory names

harbar20 committed:
removed config.json from .gitignore and reformatted airtable api inits

harbar20 committed:
added a description to the project

harbar20 committed:
removed claimant slack email when inserting new record for each qrcode

harbar20 committed:
admins are now verified using user id rather than display name

harbar20 committed:
changed pyproject version to 1.0.0 cuz who gives af

lukec11 reviewed (approved):
Looks good enough to merge! :tada: :parrot: 

github has no `:partyparrot:` :(
